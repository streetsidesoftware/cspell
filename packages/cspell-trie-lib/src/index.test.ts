import * as lib from './index';

describe('Validate index.ts', () => {
    test('index', () => {
        expect(typeof lib.Trie).toBe('function');
    });
    test('Track changes to the API.', () => {
        expect(Object.keys(lib).sort()).toMatchSnapshot();
    });
});
