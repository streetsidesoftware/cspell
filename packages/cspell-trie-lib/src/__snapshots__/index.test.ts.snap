// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validate index.ts Track changes to the API. 1`] = `
Array [
  "CASE_INSENSITIVE_PREFIX",
  "COMPOUND",
  "COMPOUND_FIX",
  "ChildMap",
  "CompoundWordsMethod",
  "FLAG_WORD",
  "FORBID",
  "FORBID_PREFIX",
  "JOIN_SEPARATOR",
  "NORMALIZED",
  "OPTIONAL_COMPOUND",
  "OPTIONAL_COMPOUND_FIX",
  "Trie",
  "TrieBuilder",
  "WORD_SEPARATOR",
  "buildTrie",
  "buildTrieFast",
  "consolidate",
  "countNodes",
  "countWords",
  "createTriFromList",
  "createTrieRoot",
  "createWeightedMap",
  "defaultTrieOptions",
  "editDistance",
  "editDistanceWeighted",
  "findNode",
  "has",
  "hintedWalker",
  "impersonateCollector",
  "importTrie",
  "insert",
  "isCircular",
  "isDefined",
  "isWordTerminationNode",
  "iterateTrie",
  "iteratorTrieWords",
  "mapDictionaryInformationToWeightMap",
  "mergeDefaults",
  "mergeOptionalWithDefaults",
  "normalizeWord",
  "normalizeWordForCaseInsensitive",
  "normalizeWordToLowercase",
  "orderTrie",
  "parseDictionary",
  "parseDictionaryLines",
  "serializeTrie",
  "suggestionCollector",
  "trieNodeToRoot",
  "walk",
  "walker",
]
`;
